extension/src/openvic-extension/utility/StringLiteral.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <string_view>
+#include <type_traits>
+
+namespace OpenVic {
+	template<std::size_t N>
+	struct StringLiteral {
+		constexpr StringLiteral(const char (&str)[N]) {
+			std::copy_n(str, N, value);
+		}
+
+		constexpr StringLiteral(std::string_view string) {
+			std::copy_n(string.begin(), N, value);
+		}
+
+		char value[N];
+		static const constexpr std::integral_constant<std::size_t, N - 1> size {};
+
+		struct iterator {
+			using iterator_concept [[maybe_unused]] = std::contiguous_iterator_tag;
+			using difference_type = std::ptrdiff_t;
+			using element_type = char; // element_type is a reserved name that must be used in the definition
+			using pointer = element_type*;
+			using reference = element_type&;
```

Comment:
I note also that this is a non-constant iterator (you could modify the data pointed to using it), but you return it from functions marked as const. Thus you are creating a potential const "leak"